mantle/kola/tests/misc/network.go

```diff
@@ -172,3 +185,247 @@ func NetworkInitramfsSecondBoot(c cluster.TestCluster) {
+                           802-3-ethernet.cloned-mac-address $mac
+  
+            # make bond
+            nmcli conn add type ovs-port conn.interface bond0 master brcnv ovs-port.bond-mode balance-slb
+            nmcli conn add type ethernet conn.interface $default_device master bond0
+            nmcli conn add type ethernet conn.interface $secondary_device master bond0
+            nmcli conn down "$profile_name" || true
+            nmcli conn mod "$profile_name" connection.autoconnect no || true
+            nmcli conn down "$secondary_profile_name" || true
+            nmcli conn mod "$secondary_profile_name" connection.autoconnect no || true
+            if ! nmcli conn up brcnv-iface; then
+                nmcli conn up "$profile_name" || true
+                nmcli conn mod "$profile_name" connection.autoconnect yes
+                nmcli conn up "$secondary_profile_name" || true
+                nmcli conn mod "$secondary_profile_name" connection.autoconnect yes
+                nmcli c delete $(nmcli c show |grep ovs-cnv |awk '{print $1}') || true
+            else
+                nmcli conn mod brcnv-iface connection.autoconnect yes
+            fi
+          else
+              echo "ovs bridge already present"
+          fi
+`
+)
+
+// NetworkSecondaryNics verifies that secondary NICs are created on the node
+func NetworkSecondaryNics(c cluster.TestCluster) {
+	primaryMac := "52:55:00:d1:56:00"
+	secondaryMac := "52:55:00:d1:56:01"
+
+	setupMultipleNetworkTest(c, primaryMac, secondaryMac)
+
+	checkOvsBridge(c, primaryMac)
+}
+
+func checkOvsBridge(c cluster.TestCluster, setPrimaryMac string) {
+	var err error
+	machineIdx := 0
+
+	ovsBridgeName := "brcnv-iface"
+	bondInterfaceMACAddress, err := getBondIfaceMAC(c, machineIdx, ovsBridgeName)
+	if err != nil {
+		c.Fatalf("failed to fetch bond MAC Address: %v", err)
+	}
+
+	// check bond interface got the primary MAC
+	if bondInterfaceMACAddress != setPrimaryMac {
+		c.Fatalf("ovs-bridge %s primary MAC %s does not match expected IP", ovsBridgeName, bondInterfaceMACAddress, setPrimaryMac)
+	}
+}
+
+func getBondIfaceMAC(c cluster.TestCluster, machineIdx int, interfaceName string) (string, error) {
+	m := c.Machines()[machineIdx]
+
+	output := string(c.MustSSH(m, fmt.Sprintf("nmcli -g 802-3-ethernet.cloned-mac-address connection show %s", interfaceName)))
+	output = strings.Replace(output, "\\:", ":", -1)
+
+	var macAddress net.HardwareAddr
+	var err error
+	if macAddress, err = net.ParseMAC(output); err != nil {
+		return "", fmt.Errorf("failed to parse MAC address %v for interface Name %s: %v", output, interfaceName, err)
+	}
+
+	return macAddress.String(), nil
+}
+
+func addKernelArgs(c cluster.TestCluster, m platform.Machine, args []string) {
+	if len(args) == 0 {
+		return
+	}
+
+	rpmOstreeCommand := "sudo rpm-ostree kargs"
+	for _, arg := range args {
+		rpmOstreeCommand = fmt.Sprintf("%s --append %s", rpmOstreeCommand, arg)
+	}
```

Comment:
Minor/optional: I think this is more legible:

```go
for _, arg := range args {
    rpmOstreeCommand += fmt.Sprintf("--append %s", arg)
}
```

Hmm, actually `strings.Join()` gives us almost what we want here. I think something like:

```go
rpmOstreeCommand := "sudo rpm-ostree kargs --append " + strings.Join(args, " --append ")
```

?

Though I agree it's less legible, so nevermind! :)